sources/osg/LightSource.js

```diff
@@ -18,7 +21,25 @@ LightSource.prototype = MACROUTILS.objectLibraryClass( MACROUTILS.objectInherit(
     },
     setLight: function ( light ) {
         this._light = light;
+    },
+    setReferenceFrame: function ( value ) {
+        this._referenceFrame = value;
+    },
+    getReferenceFrame: function () {
+        return this._referenceFrame;
+    },
+    computeBound: function ( bsphere ) {
+        Node.prototype.computeBound.call( this, bsphere );
+        if ( this._light !== undefined && this._referenceFrame === TransformEnums.RELATIVE_RF ) {
+            var position = this._light.getPosition();
+            if ( position[ 3 ] !== 0.0 ) {
+                var div = 1.0 / position[ 3 ];
+                bsphere.expandByVec3( Vec3.createAndSet( position[ 0 ] * div, position[ 1 ] * div, position[ 2 ] * div ) );
```

Comment:
Hey @Kuranes, 

Just picked the code from OSG https://github.com/openscenegraph/OpenSceneGraph/blob/master/src/osg/LightSource.cpp#L61. 

What I understand is: if the w homogeneus coordinate(p[3]) is zero the light is at infinity, if its 1 then the light is in the world space, so to get the real coordinates (x,y,z ) all of them must be divided by w. 

Then we calculate the bounding sphere using this values, just to know when the lightSource should be traversed.